### ISSUE TITLE
[Bug]: Shift click doesn't select all books in between

### ISSUE THREAD

**Issue by kanjieater** [advplyr/audiobookshelf#2950]
### What happened?

I was going to select a large group of audiobooks to work around the rate limiting:
https://github.com/advplyr/audiobookshelf/issues/1570

So i tried to use this feature
https://github.com/advplyr/audiobookshelf/issues/1020

But when you scroll it seems to revert back to single click instead of selecting the range between the items clicked. 

In addition, The feature also doesn't work on the search results, where I tried to select a range of results.

### What did you expect to happen?

Select all the books in a range with the shift modifier key pressed and a click, https://github.com/advplyr/audiobookshelf/issues/1020

### Steps to reproduce the issue

1. Hold shift and click a book
2. scroll down a few hundred books
3. Click the last book
4. it won't select the books in between

### Audiobookshelf version

v2.9.0

### How are you running audiobookshelf?

Docker

### What OS is your Audiobookshelf server hosted from?

Linux

### If the issue is being seen in the UI, what browsers are you seeing the problem on?

Chrome

**Comment by advplyr** [comment]
I can confirm that it doesn't work on the search page. It is working on the library page when scrolling for me though. Can you test that again?

Also, you don't need to hold shift when selecting the first book. You only need to hold shift when selecting the second book.

**Comment by kanjieater** [comment]
> I can confirm that it doesn't work on the search page. It is working on the library page when scrolling for me though. Can you test that again?
> 
> Also, you don't need to hold shift when selecting the first book. You only need to hold shift when selecting the second book.

Thanks. The holding or not holding is good to know. I could still reproduce the bug though. 

I made a gif of the issue, but unfortunately it's too big for me to upload anywhere 😅 It's available here for 8 hours it says
https://s51-hzfi.freeconvert.com/task/663d8ae57614e60f803142dc/chrome_j4bJhogftU.gif

**Comment by advplyr** [comment]
I was able to reproduce it by clicking the scrollbar and going to the bottom like you did in the video. 

I'm not sure how to solve this since the bookshelf is only loading in items as they become visible. We can't select an item that we don't even have an ID for.

I don't think it will be supported so we'll need to show an alert or something to let the user know.

**Comment by kanjieater** [comment]
Oh ok interesting. When you click the last element in the list, would it be possible to fetch everything in between on that interaction?

Alternatively a "Crtl+A" style option that just selects everything could mostly do the same thing

**Comment by advplyr** [comment]
Yeah but depending on the amount of items that need to be fetched this can take time and so the UI has to be updated to reflect that it is pulling all the items. Then we don't necessarily want to load that much data into the browser if you have 100k items for example. To support this I think we would need to build the bookshelf in a different way.

**Comment by v4u6h4n** [comment]
Just wondering if mass item selection (holding shift) could be added to podcast episode selection?

**Comment by TobiasDax** [comment]
Adding my voice to this. not being able to select multiple podcast episodes is a horrible oversight.
